packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesModule.kt

```diff
@@ -68,143 +68,129 @@ class UpdatesModule : Module(), IUpdatesEventManagerObserver {
     }
 
     AsyncFunction("reload") { promise: Promise ->
-      UpdatesController.instance.relaunchReactApplicationForModule(
-        object : IUpdatesController.ModuleCallback<Unit> {
-          override fun onSuccess(result: Unit) {
-            promise.resolve(null)
-          }
-
-          override fun onFailure(exception: CodedException) {
-            promise.reject(exception)
-          }
+      moduleScope.launch {
+        runCatching {
+          UpdatesController.instance.relaunchReactApplicationForModule()
+          promise.resolve()
```

Comment:
```suggestion
          promise.resolve(null)
```

maybe keep original behavior

Reply:
This is the same. We provide null is the user doesn't https://github.com/expo/expo/blob/07f4c6cbd9da1eae04b3736aada3cd0d44288a12/packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/Promise.kt#L12

Reply:
good to know, thanks!